src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { getUser } from '../../api';
+
+import { Todo } from '../../types/Todo';
+import { TodoModal } from '../TodoModal';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
```

Comment:
It's recommended to memoize the `showModal` function with `useCallback` to prevent unnecessary re-renders and improve performance.